client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -141,7 +142,8 @@ public WriteBufferManager(
     this.requireMemoryRetryMax = bufferManagerOptions.getRequireMemoryRetryMax();
     this.arrayOutputStream = new WrappedByteArrayOutputStream(serializerBufferSize);
     // in columnar shuffle, the serializer here is never used
-    if (serializer != null) {
+    this.isRowBased = (serializer != null);
```

Comment:
Is it common case? if we support other columnar shuffle like rapids, the other columnar shuffle may use non-null serializer.

Reply:
Can columnar shuffle be supported by adding configuration or adding a new constructor? This judgment seems a bit hard

Reply:
Good point

Reply:
In my opinion, all columnar shuffle should  pass null serializer to `WriterBufferManager`.
1. `WriterBufferManager` should only do the buffer related work as it's name mean
2. Columnar data framework is various, the integration work should be done in  the implementation of `rss columnar shuffle writer`

Reply:
We can pass a non-serializer  and ColumnarRowBatch to `addRecord` method although we use columnar shuffle.  It's not equivalent.

Reply:
> We can pass a non-serializer and ColumnarBatch to `addRecord` method although we use columnar shuffle. It's not equivalent.

Not only serialization but also partitioning should be handled and are both related with the implementation of the third party columnar framework, we'd better handle them outside of `WriterBufferManager`

Reply:
> Is is possible for gluten to pass additional conf items to spark conf and in Uniffle side we can add a `columnarSupport` field in the BufferManagerOptions class.

Maybe it's not a good idea to change the spark conf, since the columnar shuffle maybe fallback to row based shuffle

Reply:
> > We can pass a non-serializer and ColumnarBatch to `addRecord` method although we use columnar shuffle. It's not equivalent.
> 
> Not only serialization but also partitioning should be handled and are both related with the implementation of the third party columnar framework, we'd better handle them outside of `WriterBufferManager`

For a shuffle, we can handle partitioning by the `partitioner`. You mean that partitioner is null for gluten, won't it? 
In another word, we may use null serializer row based shuffle.

Reply:
@jerqi  When in columnar shuffle, an element of the iterator is a  `ColumnarBatch`, it's consists of many rows which will be  partitioned to different partitionIds, current spark partitioner api cannot handle this scenario.
Actually the java partitioner is fake  when use columnar shuffle in gluten, partitioner is implemented in cpp layer

Reply:
Could we have null serializer for row based shuffle in the future?

Reply:
How about indicate row based shuffle  false  in `rssConf` @jerqi

Reply:
Ok for me.